packages/atomic/src/components/atomic-query-summary/atomic-query-summary.tsx

```diff
@@ -27,77 +38,97 @@ import {
 
   @BindStateToController('querySummary')
   @State()
   private querySummaryState!: QuerySummaryState;
+  @BindStateToI18n()
+  @State()
+  private strings = {
+    noResults: () => this.bindings.i18n.t('noResults'),
+    noResultsFor: (query: string) =>
+      this.bindings.i18n.t('noResultsFor', {
+        ...this.unescapeStringOption,
+        query,
+      }),
+    showingResultsOf: (resultOfOptions: ResultOfOptions) =>
+      this.bindings.i18n.t('showingResultsOf', {
+        ...this.unescapeStringOption,
+        ...resultOfOptions,
+      }),
+    showingResultsOfWithQuery: (resultOfOptions: ResultOfOptions) =>
+      this.bindings.i18n.t('showingResultsOfWithQuery', {
+        ...this.unescapeStringOption,
+        ...resultOfOptions,
+      }),
+    inSeconds: (count: number) => this.bindings.i18n.t('inSeconds', {count}),
+  };
   @State() public error!: Error;
 
+  /**
+   * Whether to display the duration of the last query execution.
+   */
+  @Prop() enableDuration = true;
+
   public initialize() {
     this.querySummary = buildQuerySummary(this.bindings.engine);
   }
 
-  private renderNoResults() {
-    return <span part="no-result">No results{this.renderQuery()}</span>;
-  }
-
-  private renderHasResults() {
-    // TODO: This whole render loop will not work with localization
-    return [this.renderResults(), this.renderQuery(), this.renderDuration()];
-  }
-
-  private renderResults() {
-    return (
-      <span part="results">
-        Results {this.renderHighlight(this.range)} of{' '}
-        {this.renderHighlight(this.total)}
-      </span>
-    );
-  }
-
-  private get range() {
-    return `${this.querySummaryState.firstResult.toLocaleString()}-${this.querySummaryState.lastResult.toLocaleString()}`;
-  }
-
-  private get total() {
-    return this.querySummaryState.total.toLocaleString();
-  }
-
-  private renderQuery() {
-    if (this.querySummaryState.hasQuery) {
+  private renderDuration() {
+    if (this.enableDuration && this.querySummaryState.hasDuration) {
       return (
-        <span part="query">
-          {' '}
-          for {this.renderHighlight(this.querySummaryState.query)}
+        <span part="duration">
+          &nbsp;{' '}
+          {this.strings.inSeconds(this.querySummaryState.durationInSeconds)}
         </span>
       );
     }
+  }
 
-    return '';
+  private wrapHighlight(content: string) {
+    return `<b part="highlight">${content}</b>`;
   }
 
-  private renderDuration() {
-    if (this.querySummaryState.hasDuration) {
-      return (
-        <span part="duration">
-          {' '}
-          in {this.querySummaryState.durationInSeconds.toLocaleString()} seconds
-        </span>
-      );
-    }
+  private renderNoResults() {
+    const content = this.querySummaryState.hasQuery
+      ? this.strings.noResultsFor(
+          this.wrapHighlight(this.querySummaryState.query)
+        )
+      : this.strings.noResults();
+    return <span part="no-results" innerHTML={content}></span>;
```

Comment:
Since the query can be set to any string, we should escape the query here to prevent XSS attacks.

Same for line 124 below,